lib/modules/manager/droneci/extract.ts

```diff
@@ -10,31 +10,39 @@ export function extractPackageFile(content: string): PackageFile | null {
     for (let lineNumber = 0; lineNumber < lines.length; lineNumber += 1) {
       const line = lines[lineNumber];
 
-      const first_line_match = regEx(/^\s* image:\s*(['"]([^\s'"]+)\\)$/).exec(
+      const firstLineMatch = regEx(/^(\s* image:\s*)(['"]([^\s'"]+)\\)$/).exec(
         line
       );
-      if (first_line_match) {
-        let currentFrom = first_line_match[2];
-        let replaceString = first_line_match[1];
+
+      if (firstLineMatch) {
+        let currentFrom = firstLineMatch[3];
+        let replaceString = firstLineMatch[2];
 
         for (let i = lineNumber + 1; i < lines.length; i += 1) {
-          const internal_line = lines[i];
-          const middle_line_match =
-            regEx(/^(\s*([^\s'"]+)\\)$/).exec(internal_line);
-          if (middle_line_match) {
-            currentFrom += middle_line_match[2];
-            replaceString += '\n' + middle_line_match[1];
+          const internalLine = lines[i];
+          const middleLineMatch =
+            regEx(/^(\s*([^\s'"]+)\\)$/).exec(internalLine);
+          if (middleLineMatch) {
+            currentFrom += middleLineMatch[2];
+            replaceString += '\n' + middleLineMatch[1];
           } else {
-            const final_line_match = regEx(/^(\s*([^\s'"]+)['"])$/).exec(
-              internal_line
+            const finalLineMatch = regEx(/^(\s*([^\s'"]+)['"])$/).exec(
+              internalLine
             );
-            if (final_line_match) {
-              currentFrom += final_line_match[2];
-              replaceString += '\n' + final_line_match[1];
+            if (finalLineMatch) {
+              currentFrom += finalLineMatch[2];
+              replaceString += '\n' + finalLineMatch[1];
 
               const dep = getDep(currentFrom);
               dep.depType = 'docker';
               dep.replaceString = replaceString;
+              if (dep.autoReplaceStringTemplate) {
```

Comment:
i think we should push this to the docker function and pass a boolean when newlines are allowed. so we can easily support this on all yaml based managers.

@rarkins WDYT?

Reply:
The sharing part I agree. Moving it to the docker extract does not seem right though. In general the whole code reuse there seems a bit off. For example the field `currentFrom` is obviously semantic dependency to dockerfile. No other docker reference is actually `from` reference.

Additionally why docker should know anything about yml, json or any other data format. I see a clear need for data format abstraction that should handle such specifics.

I would prefer please, if you guys handle the burthen of such design decision instead of requesting contributors to combine such design work in a small feature PRs.